### ISSUE TITLE
how to compile libarchive-3.2.2 i686-w64-mingw32.static target?

### ISSUE THREAD

**Issue by yjdwbj** [mxe/mxe#1842]
Hi,all.

I found [libarchive](https://github.com/libarchive/libarchive/tree/v3.2.2) support for encrypted Zip archives,mininum version is 3.2.2 , and How to build it with i686-w64-mingw32.static target?

```bash
Currently, the library automatically detects and reads the following fomats:

Old V7 tar archives
POSIX ustar
GNU tar format (including GNU long filenames, long link names, and sparse files)
Solaris 9 extended tar format (including ACLs)
POSIX pax interchange format
POSIX octet-oriented cpio
SVR4 ASCII cpio
POSIX octet-oriented cpio
Binary cpio (big-endian or little-endian)
ISO9660 CD-ROM images (with optional Rockridge or Joliet extensions)
ZIP archives (with uncompressed or "deflate" compressed entries, including support for encrypted Zip archives)
GNU and BSD 'ar' archives
'mtree' format
7-Zip archives
Microsoft CAB format
LHA and LZH archives
RAR archives (with some limitations due to RAR's proprietary status)
XAR archives
```

**Comment by tonytheodore** [comment]
Try updating it with:
```
make update-package-libarchive
```

that will fetch the latest version.

**Comment by yjdwbj** [comment]
@tonytheodore   compile libarchive-3.3.2 will got this error
```bash
 CC       libarchive/archive_digest.lo
  CC       libarchive/archive_entry.lo
In file included from libarchive/archive_cryptor.c:32:0:
libarchive/archive_cryptor_private.h:67:20: fatal error: Bcrypt.h: No such file or directory
compilation terminated.
  CC       libarchive/archive_entry_copy_stat.lo
Makefile:5681: recipe for target 'libarchive/archive_cryptor.lo' failed
make[3]: *** [libarchive/archive_cryptor.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/home/lcy/mxe-qt5.6/tmp-libarchive-i686-w64-mingw32.static/libarchive-3.3.2'
Makefile:2969: recipe for target 'all' failed
make[2]: *** [all] Error 2
make[2]: Leaving directory '/home/lcy/mxe-qt5.6/tmp-libarchive-i686-w64-mingw32.static/libarchive-3.3.2'
Makefile:604: recipe for target 'build-only-libarchive_i686-w64-mingw32.static' failed
make[1]: *** [build-only-libarchive_i686-w64-mingw32.static] Error 2
make[1]: Leaving directory '/home/lcy/mxe-qt5.6'

real	0m22.018s
user	0m16.376s
sys	0m1.640s
```
but If I was manual compile libarchive-3.2.2    successed.

**Comment by tonytheodore** [comment]
See https://github.com/mxe/mxe/pull/1843/commits/dbfbbb0fa4eb7a1c797d45da1539233302634694

**Comment by tonytheodore** [comment]
fixed in #1843
